builder/relinker/relinker.py

```
@@ -0,0 +1,370 @@
+        else:
+            self.secs = filter_secs(secs, ('.iram1.', '.text.', '.literal.'))
+        self.isecs = strip_secs(self.secs, self.fsecs)
+
+    def __str__(self):
+        s = 'lib=%s\nfile=%s\nlib_path=%s\ndesc=%s\nsecs=%s\nfsecs=%s\nisecs=%s\n'%(\
+            self.lib, self.file, self.lib_path, self.desc, self.secs, self.fsecs,\
+            self.isecs)
+        return s
+
+class relink_c:
+    def __init__(self, input, library_file, object_file, function_file, sdkconfig_file, missing_function_info):
```

Ruff (0.15.7)

[error] Function argument `input` is shadowing a Python builtin

(A002)